### PR TITLE
Migrate to operator

### PR DESCRIPTION
**Pull request by ahmedwaleedmalik**
- [x] Convert IngressMonitor controller to an operator
- [x] Implement IngressMonitor CRD
- [x] Add support for route and ingress reference in CRD
- [x] Add generic provider specific annotations as configuration in IngressMonitor CRD
- [x] Return response from provider in case of failure for CRUD operations on monitors
- [x] Use secret or CRD with secret reference for providing IMC configuration instead of volume mount
- [x] Make `gofmt` and `golangci lint` part of the makeFile
- [x] Add basic validation for CR
- [x] Handle global configuration (https://github.com/stakater/IngressMonitorController/blob/master/pkg/config/config.go#L14)
- [x] Remove support for annotations
- [x] Refactor RBAC 
- [x] Add document for migration guide from controller(release-v1.0.x) to operator(master)
- [x] Refactor documentation
- [x] Refactor and add test cases

Things that are out of scope of this PR(will be covered in follow up issues):
- Separate CR for config
- Add status MonitorAvailable in CR
- Add provider in CR and handle CRUD actions accordingly
- Add metrics
- Create Helm Chart
- Add script to facilitate migration
- Changes required for the pipeline

### CODE REVIEW

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Yikes! You better fix it before anyone else finds out! [Build 1](https://jenkins-delivery.stakater.com/job/Stakater/job/IngressMonitorController/job/PR-262/1/) has Failed!

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Yikes! You better fix it before anyone else finds out! [Build 2](https://jenkins-delivery.stakater.com/job/Stakater/job/IngressMonitorController/job/PR-262/2/) has Failed!

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Yikes! You better fix it before anyone else finds out! [Build 3](https://jenkins-delivery.stakater.com/job/Stakater/job/IngressMonitorController/job/PR-262/3/) has Failed!

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Yikes! You better fix it before anyone else finds out! [Build 4](https://jenkins-delivery.stakater.com/job/Stakater/job/IngressMonitorController/job/PR-262/4/) has Failed!

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Image is available for testing. ``docker pull stakater/ingressmonitorcontroller:SNAPSHOT-PR-262-5``

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Image is available for testing. ``docker pull stakater/ingressmonitorcontroller:SNAPSHOT-PR-262-6``

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Image is available for testing. ``docker pull stakater/ingressmonitorcontroller:SNAPSHOT-PR-262-7``

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Yikes! You better fix it before anyone else finds out! [Build 8](https://jenkins-delivery.stakater.com/job/Stakater/job/IngressMonitorController/job/PR-262/8/) has Failed!

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Yikes! You better fix it before anyone else finds out! [Build 9](https://jenkins-delivery.stakater.com/job/Stakater/job/IngressMonitorController/job/PR-262/9/) has Failed!

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Yikes! You better fix it before anyone else finds out! [Build 10](https://jenkins-delivery.stakater.com/job/Stakater/job/IngressMonitorController/job/PR-262/10/) has Failed!

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Yikes! You better fix it before anyone else finds out! [Build 11](https://jenkins-delivery.stakater.com/job/Stakater/job/IngressMonitorController/job/PR-262/11/) has Failed!

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Yikes! You better fix it before anyone else finds out! [Build 12](https://jenkins-delivery.stakater.com/job/Stakater/job/IngressMonitorController/job/PR-262/12/) has Failed!

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Yikes! You better fix it before anyone else finds out! [Build 13](https://jenkins-delivery.stakater.com/job/Stakater/job/IngressMonitorController/job/PR-262/13/) has Failed!

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Yikes! You better fix it before anyone else finds out! [Build 14](https://jenkins-delivery.stakater.com/job/Stakater/job/IngressMonitorController/job/PR-262/14/) has Failed!

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Yikes! You better fix it before anyone else finds out! [Build 15](https://jenkins-delivery.stakater.com/job/Stakater/job/IngressMonitorController/job/PR-262/15/) has Failed!

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Yikes! You better fix it before anyone else finds out! [Build 16](https://jenkins-delivery.stakater.com/job/Stakater/job/IngressMonitorController/job/PR-262/16/) has Failed!

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Yikes! You better fix it before anyone else finds out! [Build 17](https://jenkins-delivery.stakater.com/job/Stakater/job/IngressMonitorController/job/PR-262/17/) has Failed!

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Yikes! You better fix it before anyone else finds out! [Build 18](https://jenkins-delivery.stakater.com/job/Stakater/job/IngressMonitorController/job/PR-262/18/) has Failed!

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Yikes! You better fix it before anyone else finds out! [Build 20](https://jenkins-delivery.stakater.com/job/Stakater/job/IngressMonitorController/job/PR-262/20/) has Failed!

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Yikes! You better fix it before anyone else finds out! [Build 21](https://jenkins-delivery.stakater.com/job/Stakater/job/IngressMonitorController/job/PR-262/21/) has Failed!

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Yikes! You better fix it before anyone else finds out! [Build 22](https://jenkins-delivery.stakater.com/job/Stakater/job/IngressMonitorController/job/PR-262/22/) has Failed!

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Yikes! You better fix it before anyone else finds out! [Build 23](https://jenkins-delivery.stakater.com/job/Stakater/job/IngressMonitorController/job/PR-262/23/) has Failed!

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Yikes! You better fix it before anyone else finds out! [Build 25](https://jenkins-delivery.stakater.com/job/Stakater/job/IngressMonitorController/job/PR-262/25/) has Failed!

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Yikes! You better fix it before anyone else finds out! [Build 26](https://jenkins-delivery.stakater.com/job/Stakater/job/IngressMonitorController/job/PR-262/26/) has Failed!

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Yikes! You better fix it before anyone else finds out! [Build 27](https://jenkins-delivery.stakater.com/job/Stakater/job/IngressMonitorController/job/PR-262/27/) has Failed!

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Yikes! You better fix it before anyone else finds out! [Build 28](https://jenkins-delivery.stakater.com/job/Stakater/job/IngressMonitorController/job/PR-262/28/) has Failed!

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Yikes! You better fix it before anyone else finds out! [Build 29](https://jenkins-delivery.stakater.com/job/Stakater/job/IngressMonitorController/job/PR-262/29/) has Failed!

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Yikes! You better fix it before anyone else finds out! [Build 30](https://jenkins-delivery.stakater.com/job/Stakater/job/IngressMonitorController/job/PR-262/30/) has Failed!

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Image is available for testing. ``docker pull stakater/ingressmonitorcontroller:SNAPSHOT-PR-262-31``

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Image is available for testing. ``docker pull stakater/ingressmonitorcontroller:SNAPSHOT-PR-262-32``

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Image is available for testing. ``docker pull stakater/ingressmonitorcontroller:SNAPSHOT-PR-262-33``

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Image is available for testing. ``docker pull stakater/ingressmonitorcontroller:SNAPSHOT-PR-262-34``

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Image is available for testing. ``docker pull stakater/ingressmonitorcontroller:SNAPSHOT-PR-262-36``

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Yikes! You better fix it before anyone else finds out! [Build 37](https://jenkins-delivery.stakater.com/job/Stakater/job/IngressMonitorController/job/PR-262/37/) has Failed!

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Image is available for testing. ``docker pull stakater/ingressmonitorcontroller:SNAPSHOT-PR-262-38``

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Image is available for testing. ``docker pull stakater/ingressmonitorcontroller:SNAPSHOT-PR-262-39``

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Image is available for testing. ``docker pull stakater/ingressmonitorcontroller:SNAPSHOT-PR-262-40``

**Comment by ahmedwaleedmalik** [comment]
@waseem-h @hazim1093 This PR is good for an initial review.

Things that are out of scope of this PR(will be covered in follow up issues):
- Separate CR for config
- Add status MonitorAvailable in CR
- Add provider in CR and handle CRUD actions accordingly
- Add metrics

Currently, following are the known issues in this PR(apart from the sub task list mentioned in PR description):
- Add basic validation for CR
- For monitor/CR update, Compare state of monitors before sending an API call 
- Handle global configuration (https://github.com/stakater/IngressMonitorController/blob/master/pkg/config/config.go#L14)

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Image is available for testing. ``docker pull stakater/ingressmonitorcontroller:SNAPSHOT-PR-262-41``

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Image is available for testing. ``docker pull stakater/ingressmonitorcontroller:SNAPSHOT-PR-262-42``

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Image is available for testing. ``docker pull stakater/ingressmonitorcontroller:SNAPSHOT-PR-262-47``

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Image is available for testing. ``docker pull stakater/ingressmonitorcontroller:SNAPSHOT-PR-262-48``

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Image is available for testing. ``docker pull stakater/ingressmonitorcontroller:SNAPSHOT-PR-262-49``

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Image is available for testing. ``docker pull stakater/ingressmonitorcontroller:SNAPSHOT-PR-262-50``

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Image is available for testing. ``docker pull stakater/ingressmonitorcontroller:SNAPSHOT-PR-262-51``

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Image is available for testing. ``docker pull stakater/ingressmonitorcontroller:SNAPSHOT-PR-262-52``

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Yikes! You better fix it before anyone else finds out! [Build 53](https://jenkins-delivery.stakater.com/job/Stakater/job/IngressMonitorController/job/PR-262/53/) has Failed!

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Yikes! You better fix it before anyone else finds out! [Build 54](https://jenkins-delivery.stakater.com/job/Stakater/job/IngressMonitorController/job/PR-262/54/) has Failed!

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Image is available for testing. ``docker pull stakater/ingressmonitorcontroller:SNAPSHOT-PR-262-55``

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Image is available for testing. ``docker pull stakater/ingressmonitorcontroller:SNAPSHOT-PR-262-56``

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Image is available for testing. ``docker pull stakater/ingressmonitorcontroller:SNAPSHOT-PR-262-57``

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Image is available for testing. ``docker pull stakater/ingressmonitorcontroller:SNAPSHOT-PR-262-58``

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Image is available for testing. ``docker pull stakater/ingressmonitorcontroller:SNAPSHOT-PR-262-59``

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Image is available for testing. ``docker pull stakater/ingressmonitorcontroller:SNAPSHOT-PR-262-60``

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Image is available for testing. ``docker pull stakater/ingressmonitorcontroller:SNAPSHOT-PR-262-61``

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Image is available for testing. ``docker pull stakater/ingressmonitorcontroller:SNAPSHOT-PR-262-62``

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Image is available for testing. ``docker pull stakater/ingressmonitorcontroller:SNAPSHOT-PR-262-63``

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Image is available for testing. ``docker pull stakater/ingressmonitorcontroller:SNAPSHOT-PR-262-64``

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Image is available for testing. ``docker pull stakater/ingressmonitorcontroller:SNAPSHOT-PR-262-65``

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Image is available for testing. ``docker pull stakater/ingressmonitorcontroller:SNAPSHOT-PR-262-66``

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Image is available for testing. ``docker pull stakater/ingressmonitorcontroller:SNAPSHOT-PR-262-67``

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Image is available for testing. ``docker pull stakater/ingressmonitorcontroller:SNAPSHOT-PR-262-68``

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Image is available for testing. ``docker pull stakater/ingressmonitorcontroller:SNAPSHOT-PR-262-69``

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Image is available for testing. ``docker pull stakater/ingressmonitorcontroller:SNAPSHOT-PR-262-70``

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Image is available for testing. ``docker pull stakater/ingressmonitorcontroller:SNAPSHOT-PR-262-71``

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Image is available for testing. ``docker pull stakater/ingressmonitorcontroller:SNAPSHOT-PR-262-72``

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Image is available for testing. ``docker pull stakater/ingressmonitorcontroller:SNAPSHOT-PR-262-73``

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Image is available for testing. ``docker pull stakater/ingressmonitorcontroller:SNAPSHOT-PR-262-74``

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Image is available for testing. ``docker pull stakater/ingressmonitorcontroller:SNAPSHOT-PR-262-75``

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Yikes! You better fix it before anyone else finds out! [Build 76](https://jenkins-delivery.stakater.com/job/Stakater/job/IngressMonitorController/job/PR-262/76/) has Failed!

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Yikes! You better fix it before anyone else finds out! [Build 77](https://jenkins-delivery.stakater.com/job/Stakater/job/IngressMonitorController/job/PR-262/77/) has Failed!

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Yikes! You better fix it before anyone else finds out! [Build 78](https://jenkins-delivery.stakater.com/job/Stakater/job/IngressMonitorController/job/PR-262/78/) has Failed!

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Yikes! You better fix it before anyone else finds out! [Build 79](https://jenkins-delivery.stakater.com/job/Stakater/job/IngressMonitorController/job/PR-262/79/) has Failed!

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Yikes! You better fix it before anyone else finds out! [Build 80](https://jenkins-delivery.stakater.com/job/Stakater/job/IngressMonitorController/job/PR-262/80/) has Failed!

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Yikes! You better fix it before anyone else finds out! [Build 81](https://jenkins-delivery.stakater.com/job/Stakater/job/IngressMonitorController/job/PR-262/81/) has Failed!

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Yikes! You better fix it before anyone else finds out! [Build 82](https://jenkins-delivery.stakater.com/job/Stakater/job/IngressMonitorController/job/PR-262/82/) has Failed!

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Yikes! You better fix it before anyone else finds out! [Build 83](https://jenkins-delivery.stakater.com/job/Stakater/job/IngressMonitorController/job/PR-262/83/) has Failed!

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Yikes! You better fix it before anyone else finds out! [Build 85](https://jenkins-delivery.stakater.com/job/Stakater/job/IngressMonitorController/job/PR-262/85/) has Failed!

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Yikes! You better fix it before anyone else finds out! [Build 87](https://jenkins-delivery.stakater.com/job/Stakater/job/IngressMonitorController/job/PR-262/87/) has Failed!

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Yikes! You better fix it before anyone else finds out! [Build 88](https://jenkins-delivery.stakater.com/job/Stakater/job/IngressMonitorController/job/PR-262/88/) has Failed!

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Image is available for testing. ``docker pull stakater/ingressmonitorcontroller:SNAPSHOT-PR-262-90``

File: pkg/monitors/monitor-service.go [hunk 1]
```diff
@@ -12,10 +14,54 @@ type MonitorService interface {
...
+
+	return monitorServices
+}
+
+func SetupMonitorServicesForProvidersTest(providers []config.Provider) []MonitorServiceProxy {
+	if len(providers) < 1 {
+		log.Panic("Cannot Instantiate controller with no providers")
+	}
+	// TODO: Fix provider specific implementation and then add them to this list
+	allowedProviders := []string{"UptimeRobot", "StatusCake"}
```
Comment (hazim1093) [inline pkg/monitors/monitor-service.go:46]: Are the rest providers not implemented with the operator pattern yet?

File: pkg/monitors/monitor-service.go [hunk 2]
```diff
@@ -12,10 +14,54 @@ type MonitorService interface {
...
+
+	monitorServices := []MonitorServiceProxy{}
+
+	for index := 0; index < len(providers); index++ {
+		if contains(allowedProviders, providers[index].Name) {
```
Comment (hazim1093) [inline pkg/monitors/monitor-service.go:51]: we should log a message for the unsupported providers at least so the user knows why it doesn't work at this stage

File: build/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,11 @@
+FROM registry.svc.ci.openshift.org/openshift/release:golang-1.10 AS builder
```
[the remark this answers is not in the document]
Comment (ahmedwaleedmalik) [reply build/Dockerfile:1]: Updated

File: deploy/operator.yaml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+      name: ingressmonitorcontroller
+  template:
+    metadata:
+      labels:
+        name: ingressmonitorcontroller
+    spec:
+      serviceAccountName: ingressmonitorcontroller
+      containers:
+        - name: ingressmonitorcontroller
+          # Replace this with the built image name
```
[the remark this answers is not in the document]
Comment (ahmedwaleedmalik) [reply deploy/operator.yaml:18]: Removed

File: go.mod [hunk 5]
```diff
@@ -3,24 +3,25 @@ module github.com/stakater/IngressMonitorController
 go 1.14
 
 require (
-	cloud.google.com/go v0.38.0
+	cloud.google.com/go v0.49.0
 	github.com/Azure/azure-sdk-for-go v44.0.0+incompatible
 	github.com/Azure/go-autorest/autorest/azure/auth v0.5.0
 	github.com/Azure/go-autorest/autorest/to v0.4.0 // indirect
```
[the remark this answers is not in the document]
Comment (ahmedwaleedmalik) [reply go.mod:9]: I did that already, just ran that again and there are no changes.

File: pkg/monitors/monitor-service.go (hunk 1 above)
Comment (ahmedwaleedmalik) [reply pkg/monitors/monitor-service.go:46]: The are implemented and integrated within the operator. They have just been disabled for this particular test case that i wrote against CRD reconciliation. For AppInsights, Pingdom and Uptime we don't have the appropriate credentials. And for Updown and GCloud there are issues with the existing implementation, that are there before the migration to operator. Had to restrict this test case to only a handful of providers.

File: pkg/monitors/monitor-service.go (hunk 2 above)
Comment (ahmedwaleedmalik) [reply pkg/monitors/monitor-service.go:51]: I'll add an additional log here although this is only applicable for test cases. Currently, only UptimeRobot and StatusCake are supported for reconciliation test cases

File: pkg/controller/endpointmonitor/endpointmonitor_controller_test.go [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+			APIVersion: "endpointmonitor.stakater.com/v1alpha1",
+		},
+		Spec: endpointmonitorv1alpha1.EndpointMonitorSpec{
+			URL:        testURL,
+			ForceHTTPS: true,
+		},
+	}
+)
+
+func TestEndpointMonitorReconcile(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (ahmedwaleedmalik) [reply pkg/controller/endpointmonitor/endpointmonitor_controller_test.go:46]: Good point, i'll split it down.

[the remark this answers is not in the document]
Comment (ahmedwaleedmalik) [reply pkg/controller/endpointmonitor/endpointmonitor_controller_test.go:46]: Done

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Image is available for testing. ``docker pull stakater/ingressmonitorcontroller:SNAPSHOT-PR-262-92``

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Yikes! You better fix it before anyone else finds out! [Build 93](https://jenkins-delivery.stakater.com/job/Stakater/job/IngressMonitorController/job/PR-262/93/) has Failed!

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Yikes! You better fix it before anyone else finds out! [Build 94](https://jenkins-delivery.stakater.com/job/Stakater/job/IngressMonitorController/job/PR-262/94/) has Failed!

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Image is available for testing. ``docker pull stakater/ingressmonitorcontroller:SNAPSHOT-PR-262-95``

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Image is available for testing. ``docker pull stakater/ingressmonitorcontroller:SNAPSHOT-PR-262-96``

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Image is available for testing. ``docker pull stakater/ingressmonitorcontroller:SNAPSHOT-PR-262-97``

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Image is available for testing. ``docker pull stakater/ingressmonitorcontroller:SNAPSHOT-PR-262-98``

File: pkg/monitors/monitor-service.go (hunk 2 above)
Comment (hazim1093) [reply pkg/monitors/monitor-service.go:53]: Oh Ok I see that it's only for test cases

File: pkg/controller/endpointmonitor/endpointmonitor_controller_test.go [hunk 7]
```diff
@@ -43,7 +43,68 @@ var (
 	}
 )
 
-func TestEndpointMonitorReconcile(t *testing.T) {
+func TestEndpointMonitorReconcileCreate(t *testing.T) {
+	controllerConfig := config.GetControllerConfigTest()
```
Comment (hazim1093) [inline pkg/controller/endpointmonitor/endpointmonitor_controller_test.go:47]: Can the common part of these tests be moved to a helper method ?

Comment (ahmedwaleedmalik) [reply pkg/controller/endpointmonitor/endpointmonitor_controller_test.go:47]: I tried doing that but they are pretty much dependent on each other. But i'll check again

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Image is available for testing. ``docker pull stakater/ingressmonitorcontroller:SNAPSHOT-PR-262-100``

File: pkg/controller/endpointmonitor/endpointmonitor_controller_test.go (hunk 7 above)
Comment (ahmedwaleedmalik) [reply pkg/controller/endpointmonitor/endpointmonitor_controller_test.go]: Moved

**Pull request merged**